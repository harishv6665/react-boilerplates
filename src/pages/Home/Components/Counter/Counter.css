.title {
  text-align: center;
  font-size: 30px;
  color: #3a3a3a;
}

.wrapper {
  height: 250px;
  margin-top: 70px;
  display: flex;
  justify-content: center;
}

.actionBtn {
  width: 250px;
  font-size: 160px;
  padding-bottom: 30px;
  font-weight: 300;
  border-radius: 50%;
  background-color: #03A9F4;
  color: #fff;
  box-shadow: 3px 3px 10px 0px #0000004d;
  border: 0;
}

.text {
  width: 350px;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 150px;
}
