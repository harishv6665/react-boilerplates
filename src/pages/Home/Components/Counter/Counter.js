import React from 'react';
import PropTypes from 'prop-types';
import styles from './Counter.css';

const Counter = ({ count, incrementCount, decrementCount }) => (
  <>
    <p className={styles.title}>Simple counter example using redux</p>
    <div className={styles.wrapper}>
      <button
        className={styles.actionBtn}
        onClick={() => decrementCount()}
      >-</button>
      <span className={styles.text}>{count}</span>
      <button
        className={styles.actionBtn}
        onClick={() => incrementCount()}
      >+</button>
    </div>
  </>
);

export default Counter;

Counter.defaultProps = {
  count: 0,
  incrementCount: f => f,
  decrementCount: f => f
};

Counter.propTypes = {
  count: PropTypes.number,
  incrementCount: PropTypes.func,
  decrementCount: PropTypes.func
};
