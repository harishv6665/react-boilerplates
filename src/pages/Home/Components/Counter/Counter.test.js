import React from 'react';
import { shallow } from 'enzyme';
import Counter from './Counter';

describe('Counter', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Counter debug />);

    expect(component).toMatchSnapshot();
  });
});