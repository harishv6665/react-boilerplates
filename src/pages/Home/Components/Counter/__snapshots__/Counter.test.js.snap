// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render correctly in "debug" mode 1`] = `
<Fragment>
  <p
    className="title"
  >
    Simple counter example using redux & redux-saga.
    <br />
    Demonstrating using a promise with 1 second timeout delay.
  </p>
  <div
    className="wrapper"
  >
    <button
      className="actionBtn"
      onClick={[Function]}
    >
      -
    </button>
    <span
      className="text"
    >
      0
    </span>
    <button
      className="actionBtn"
      onClick={[Function]}
    >
      +
    </button>
  </div>
</Fragment>
`;
